src/main/java/applemazer/Parser.java

```diff
@@ -0,0 +1,177 @@
+package applemazer;
+
+import java.util.Scanner;
+
+import commands.ByeCommand;
+import commands.Command;
+import commands.DeadlineCommand;
+import commands.EventCommand;
+import commands.FindCommand;
+import commands.IntegerCommand;
+import commands.ListCommand;
+import commands.NonExistentCommand;
+import commands.TodoCommand;
+
+/**
+ * Class that parses all user commands.
+ */
+public class Parser {
+    private final Scanner sc;
+
+    public Parser(Scanner sc) {
+        this.sc = sc;
+    }
+
+    /**
+     * Parses additional information from the user input based on the given command.
+     * @param command Command that the user inputs.
+     * @return Command object that is ready to execute the user command.
+     * @throws Exception Thrown when parsing fails due to incorrect user input format.
+     */
+    public Command parse(String command) throws Exception {
+        String desc;
+        String[] split;
+        int taskNumber;
+
+        return switch (command) {
+        case "bye" -> {
```

Comment:
##Missing Hi/Hello Command

It looks like the application might be missing a response for a `hi` or `hello` command. Since users might naturally start an interaction with a greeting, adding this command could improve user experience. For instance, a simple response like "Hi there! How can I assist you today?" would be great.